src/analysis/vector/geometry_checker/qgsvectorlayerfeaturepool.cpp

```diff
@@ -17,29 +17,14 @@ email                : matthias@opengis.ch
 #include "qgsthreadingutils.h"
 
 #include "qgsfeaturerequest.h"
+#include "qgsvectorlayer.h"
 
 QgsVectorLayerFeaturePool::QgsVectorLayerFeaturePool( QgsVectorLayer *layer )
-  : QgsFeaturePool( layer )
+  : QObject()
+  , QgsFeaturePool( layer )
 {
-  // Build spatial index
-  QgsFeature feature;
-  QgsFeatureRequest req;
-  QgsFeatureIds featureIds;
-
-  QgsFeatureIterator it = layer->getFeatures( req );
-  while ( it.nextFeature( feature ) )
-  {
-    if ( feature.geometry() )
-    {
-      insertFeature( feature );
-      featureIds.insert( feature.id() );
-    }
-    else
-    {
-      featureIds.remove( feature.id() );
-    }
-  }
-  setFeatureIds( featureIds );
+  connect( layer, &QgsVectorLayer::featureDeleted, this, &QgsVectorLayerFeaturePool::onFeatureDeleted );
```

Comment:
Have we any unit tests for QgsVectorLayerFeaturePool ? This is screaming out for some. (Especially given that we're past freeze and need to proceed with caution)

Reply:
Aaaaand the tests have landed as well :tada: